tests/test_network_model_data.py

```
@@ -182,63 +142,214 @@
+    assert any(tech.sector == 'demand' for tech in techs)
+
+
+def test_sector_handling_without_sectors():
+    """Test that sectors are handled gracefully when they don't exist in the database."""
+    # Mock database without sector column
+    mock_con = MagicMock(name='mock_connection')
+    mock_cursor = MagicMock(name='mock_cursor')
+    mock_con.cursor.return_value = mock_cursor
+
+    # Mock the sector column check to raise OperationalError (column doesn't exist)
+    def dispatcher(query: str, *_: object):
```

Ruff (0.14.1)

Missing return type annotation for private function `dispatcher`

(ANN202)